src/frontend/src/sol/schedulers/sol-wallet.scheduler.ts

```diff
@@ -0,0 +1,115 @@
+import { WALLET_TIMER_INTERVAL_MILLIS } from '$lib/constants/app.constants';
+import { SchedulerTimer, type Scheduler, type SchedulerJobData } from '$lib/schedulers/scheduler';
+import type { SolAddress } from '$lib/types/address';
+import type {
+	PostMessageDataRequestSol,
+	PostMessageDataResponseError
+} from '$lib/types/post-message';
+import type { CertifiedData } from '$lib/types/store';
+import { loadSolLamportsBalance } from '$sol/api/solana.api';
+import type { SolanaNetworkType } from '$sol/types/network';
+import type { SolPostMessageDataResponseWallet } from '$sol/types/sol-post-message';
+import { assertNonNullish, isNullish } from '@dfinity/utils';
+
+interface LoadSolWalletParams {
+	solanaNetwork: SolanaNetworkType;
+	address: SolAddress;
+}
+
+interface SolWalletStore {
+	balance: CertifiedData<bigint | null> | undefined;
+}
+
+interface SolWalletData {
+	balance: CertifiedData<bigint | null>;
+}
+
+export class SolWalletScheduler implements Scheduler<PostMessageDataRequestSol> {
+	private timer = new SchedulerTimer('syncSolWalletStatus');
+
+	private store: SolWalletStore = {
+		balance: undefined
+	};
+
+	stop() {
+		this.timer.stop();
+	}
+
+	async start(data: PostMessageDataRequestSol | undefined) {
+		await this.timer.start<PostMessageDataRequestSol>({
+			interval: WALLET_TIMER_INTERVAL_MILLIS,
+			job: this.syncWallet,
+			data
+		});
+	}
+
+	async trigger(data: PostMessageDataRequestSol | undefined) {
+		await this.timer.trigger<PostMessageDataRequestSol>({
+			job: this.syncWallet,
+			data
+		});
+	}
+
+	private loadBalance = async (
+		params: LoadSolWalletParams
+	): Promise<CertifiedData<bigint | null>> => ({
+		data: await loadSolLamportsBalance({ network: params.solanaNetwork, address: params.address }),
+		certified: true
```

Comment:
The balance you're fetching here (from 3rd party API) is not certified. Same as with Infura and ETH.

Reply:
Hm ok, what does certified mean? By default the SDK uses commitment level 'finalized', so the transaction is definite. https://solana.com/docs/rpc#default-commitment